plugins/module_utils/vmware_rest_client.py

```diff
@@ -176,7 +176,7 @@ def get_tags_for_object(self, tag_service=None, tag_assoc_svc=None, dobj=None):
         """
         # This method returns list of tag objects only,
         # Please use get_tags_for_dynamic_obj for more object details
-        tags = []
+        tags = set()
```

Comment:
Sorry for the late reply, there were so many other things to do :-/

I see a problem here. The module utils are considered part of the API of a collection. Somebody else might use it, so we can't just change the return type from list to set. At least not in a minor release, we would have to wait until version 4.

Do you use a set to speed things up only here? Then you could `return list(tags)`, this wouldn't be breaking.

Or does a set also speed up things in the modules? In that case, I would suggest a new method `get_tags_for_object_set(self, tag_service=None, tag_assoc_svc=None, dobj=None)` or similar and use this one. This would also not be breaking. Ideally, the original `get_tags_for_object` should be simplified to just `return list(self.get_tags_for_object_set(tag_service, tag_assoc_svc, dobj))` then.

Reply:
No problem, there is a lot to do these days :-).

Ok, I understand your concerns. For a quicker tag assignment, we can benefit from the fact, that we can calculate the difference between the current and the desired tag assignment more efficiently from sets instead of lists. The tag list can be really large in big environments and I would like to reduce the overhead from creating a new set from the list being returned by `get_tags_for_object_set`.

But you are right, backward compatibility is important. We can do it with a second method, what you have proposed or we avoid it and pass the collection (`list` or `set`), to which the tags should be added, to the method and return it as the result. The parameter gets initialized by default with an empty list, so that there is no difference in the default behavior of the method.

I will update the pull request, that explains much better what I mean :-).

What do you think?

Reply:
There was another breaking change in the `vmware_rest_client` that I have fixed. I added the methods `get_tag_by_category_id` and `get_tag_by_category_name` as a replacement for  `get_tag_by_category`. But, as you said, `vmware_rest_client` is part of the collection API (I was not aware of that fact), that means that we cannot remove it immediately. I have added a deprecation message to the method.